spring/spring-web-reactive/src/main/java/io/quarkus/ts/spring/web/boostrap/web/BookController.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.quarkus.hibernate.reactive.panache.Panache;
+import io.quarkus.hibernate.reactive.panache.common.runtime.ReactiveTransactional;
+import io.quarkus.ts.spring.web.boostrap.persistence.model.Book;
+import io.quarkus.ts.spring.web.boostrap.persistence.repo.BookRepository;
+import io.quarkus.ts.spring.web.boostrap.web.exception.BookIdMismatchException;
+import io.quarkus.ts.spring.web.boostrap.web.exception.BookNotFoundException;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@RestController
+@RequestMapping("/api/books")
+@Produces("application/json")
+@Consumes("application/json")
+public class BookController {
+
+    private static final int FAIL_AFTER_SECONDS = 10;
+
+    @Autowired
+    BookRepository bookRepository;
+
+    @GetMapping
+    public Uni<List<Book>> findAll() {
+        return bookRepository.listAll().onItem()
+                .ifNull().continueWith(new ArrayList<Book>())
+                .ifNoItem().after(Duration.ofSeconds(FAIL_AFTER_SECONDS)).failWith(BookNotFoundException::new);
+    }
+
+    @GetMapping("/title/{bookTitle}")
+    public Multi<Book> findByTitle(@PathVariable String bookTitle) {
+        return bookRepository.findByTitle(bookTitle)
+                .ifNoItem().after(Duration.ofSeconds(FAIL_AFTER_SECONDS)).failWith(BookNotFoundException::new);
+    }
+
+    @GetMapping("/{id}")
+    public Uni<Book> findOne(@PathVariable long id) {
+        return bookRepository.findById(id)
+                .onItem().ifNull().failWith(BookNotFoundException::new)
+                .ifNoItem().after(Duration.ofSeconds(FAIL_AFTER_SECONDS)).failWith(BookNotFoundException::new);
+    }
+
+    @PostMapping
+    public Uni<Response> create(@RequestBody Book book) {
+        return Panache.withTransaction(() -> bookRepository.persist(book))
+                .replaceWith(Response.ok((book)).status(CREATED)::build);
+    }
+
+    @DeleteMapping("/{id}")
+    @ReactiveTransactional
+    public Uni<Response> delete(@PathVariable long id) {
+        return bookRepository.deleteById(id)
+                .map(deleted -> deleted
+                        ? Response.ok().status(NO_CONTENT).build()
+                        : Response.ok().status(NOT_FOUND).build());
+    }
+
+    @PutMapping("/{id}")
+    @ReactiveTransactional
+    public Uni<Book> update(@RequestBody Book book, @PathVariable long id) {
+        if (book.getId() != id) {
+            throw new BookIdMismatchException();
+        }
+
+        return bookRepository.findById(id)
+                .onItem().ifNull().failWith(BookNotFoundException::new)
+                .onItem().ifNotNull().invoke(item -> {
+                    item.setAuthor(book.getAuthor());
+                    item.setTitle(book.getTitle());
+                    bookRepository.persist(item);
```

Comment:
I think that here we should have a return isn't?

I mean we should return the updated book
```
return bookRepository.findById(id)
                .onItem().ifNull().failWith(BookNotFoundException::new)
                .onItem().ifNotNull().call(item -> {
                    item.setAuthor(book.getAuthor());
                    item.setTitle(book.getTitle());
                    return bookRepository.persist(item);
                });
```

I am not sure if is required

Reply:
I checked the output of PUT in the test and it returns updated record